stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/GrpcServicesBuilder.java

```diff
@@ -0,0 +1,277 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.net.grpc;
+
+import com.google.protobuf.Descriptors;
+import io.grpc.MethodDescriptor;
+import io.grpc.Server;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptors;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.ServerServiceDefinition.Builder;
+import io.grpc.netty.NettyServerBuilder;
+import io.grpc.protobuf.ProtoUtils;
+import io.grpc.stub.ServerCalls;
+import io.netty.channel.nio.NioEventLoopGroup;
+import org.ballerinalang.connector.api.AnnAttrValue;
+import org.ballerinalang.connector.api.Annotation;
+import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.net.grpc.exception.GrpcServerException;
+import org.ballerinalang.net.grpc.interceptor.ServerHeaderInterceptor;
+import org.ballerinalang.net.grpc.listener.BidirectionalStreamingListener;
+import org.ballerinalang.net.grpc.listener.ClientStreamingListener;
+import org.ballerinalang.net.grpc.listener.ServerStreamingListener;
+import org.ballerinalang.net.grpc.listener.UnaryMethodListener;
+import org.ballerinalang.net.grpc.proto.ServiceProtoConstants;
+import org.ballerinalang.net.grpc.proto.ServiceProtoUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.ballerinalang.net.grpc.builder.BalGenConstants.FILE_SEPARATOR;
+
+/**
+ * This is the gRPC implementation for the {@code BallerinaServerConnector} API.
+ *
+ * @since 0.96.1
+ */
+public class GrpcServicesBuilder {
+    private io.grpc.ServerBuilder serverBuilder = null;
+    
+    void registerService(Service service) throws GrpcServerException {
+        Annotation serviceAnnotation = MessageUtils.getServiceConfigAnnotation(service, MessageConstants
+                .PROTOCOL_PACKAGE_GRPC);
+        if (serviceAnnotation != null && serviceAnnotation.getAnnAttrValue("port") != null) {
+            serverBuilder = NettyServerBuilder.forPort((int)
```

Comment:
Wouldn't it be useful move/refactor this grpc transport to carbon transport? Then it will in common place that other people could reuse.